lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -385,6 +417,7 @@ def add_app_targets
               app_native_target = AppHostInstaller.new(sandbox, project, platform, subspec_name, spec_name,
                                                        app_target_label, :add_main => false).install!
 
+              app_native_target.product_reference.name = app_target_label
```

Comment:
wonder if this could move into `AppHostInstaller` but meh.

Reply:
it seems like it should

Reply:
I was just mirroring the way we do it for test specs